src/Symfony/Component/Validator/Mapping/ElementMetadata.php

```diff
@@ -23,4 +21,10 @@
  */
 abstract class ElementMetadata extends GenericMetadata
 {
+    public function __construct()
+    {
+        if (__CLASS__ === get_class($this) || !in_array(get_parent_class($this), array('Symfony\Component\Validator\Mapping\MemberMetadata', 'Symfony\Component\Validator\Mapping\ClassMetadata'))) {
```

Comment:
And instead of ``!in_array(get_parent_class($this), array('Symfony\Component\Validator\Mapping\MemberMetadata', 'Symfony\Component\Validator\Mapping\ClassMetadata'))``, you could use ``!$this instanceof MemberMetadata && !his instanceof ClassMetadata``